### ISSUE TITLE
Startup: [Ubu IDE] GM crashes in 2024.1100.0.634 due to new IPv6/4 routing fix

### ISSUE THREAD

**Issue by SunCatMC** [YoYoGames/GameMaker-Bugs#7896]
### Description

actual ubuntu version is 24.04.1 LTS, but i can't select that as a bug report
I was able to launch and use gamemaker for a bit, but now i'm encountering a freeze instead. The launch log with all exceptions is attached
[launch log.txt](https://github.com/user-attachments/files/17258530/launch.log.txt)
I can't actually attach a care package because at this point the help button is greyed out

### Steps To Reproduce

1. Start GameMaker
2. It freezes

### How reliably can you recreate this issue using your steps above?

Always

### Which version of GameMaker are you reporting this issue for?

2024.1100 (Betas)

### Which operating system(s) are you seeing the problem on?

Ubuntu 24 LTS

### Are you running GameMaker from inside your Steam library?

None

### Sample Package Attached?

- [ ] I have attached my bug package

### Sample Project Added?

- [ ] I have included a small sample project

**Comment by YYDan** [comment]
"actual ubuntu version is 24.04.1 LTS, but i can't select that as a bug report" - fixed that now. Sorry!

**Comment by YYDan** [comment]
Same error as https://github.com/YoYoGames/GameMaker-Bugs/issues/7894

```
Checking network....
--------------------------------------------------------------------------------
Unobserved Task Exception: System.Net.Sockets.SocketException
Exception message: Network is unreachable
Exception stack:    at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.CreateException(SocketError error, Boolean forAsyncThrow)
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ConnectAsync(Socket socket)
   at System.Net.Sockets.Socket.ConnectAsync(EndPoint remoteEP, CancellationToken cancellationToken)
   at YoYoStudio.Core.Utils.HttpUtil.HappyEyeballsHttp.AttemptConnection(Int32 index, IPAddress address, Int32 port, CancellationToken cancel)
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine)
   at YoYoStudio.Core.Utils.HttpUtil.HappyEyeballsHttp.AttemptConnection(Int32 index, IPAddress address, Int32 port, CancellationToken cancel)
   at YoYoStudio.Core.Utils.HttpUtil.HappyEyeballsHttp.<>c__DisplayClass2_0.<OnConnect>b__0(Int32 i, CancellationToken cancel)
   at YoYoStudio.Core.Utils.HttpUtil.HappyEyeballsHttp.ParallelTask[T](Int32 candidateCount, Func`3 taskBuilder, TimeSpan delay, CancellationToken cancel)
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine)
   at YoYoStudio.Core.Utils.HttpUtil.HappyEyeballsHttp.ParallelTask[T](Int32 candidateCount, Func`3 taskBuilder, TimeSpan delay, CancellationToken cancel)
   at YoYoStudio.Core.Utils.HttpUtil.HappyEyeballsHttp.OnConnect(SocketsHttpConnectionContext context, CancellationToken cancellationToken)
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.AsyncStateMachineBox`1.ExecutionContextCallback(Object s)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.AsyncStateMachineBox`1.MoveNext(Thread threadPoolThread)
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.AsyncStateMachineBox`1.MoveNext()
   at System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox box, Boolean allowInlining)
   at System.Threading.Tasks.Task.RunContinuations(Object continuationObject)
   at System.Threading.Tasks.Task`1.TrySetResult(TResult result)
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetExistingTaskResult(Task`1 task, TResult result)
   at YoYoStudio.Core.Utils.HttpUtil.HappyEyeballsHttp.GetIpsForHost(DnsEndPoint endPoint, CancellationToken cancel)
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.AsyncStateMachineBox`1.ExecutionContextCallback(Object s)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.AsyncStateMachineBox`1.MoveNext(Thread threadPoolThread)
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.AsyncStateMachineBox`1.MoveNext()
   at System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox box, Boolean allowInlining)
   at System.Threading.Tasks.Task.RunContinuations(Object continuationObject)
   at System.Threading.Tasks.Task.FinishSlow(Boolean userDelegateExecute)
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
   at System.Threading.PortableThreadPool.WorkerThread.WorkerThreadStart()
--- End of stack trace from previous location ---
   at YoYoStudio.Core.Utils.HttpUtil.HappyEyeballsHttp.AttemptConnection(Int32 index, IPAddress address, Int32 port, CancellationToken cancel)
--------------------------------------------------------------------------------
```

Looks to be that your system does not like the fix for https://github.com/YoYoGames/GameMaker-Bugs/issues/7152 - Will need to investigate the original change further.

**Comment by SunCatMC** [comment]
funnily enough, on archlinux i still get that exception in console, but it doesn't stop gamemaker from functioning

**Comment by yoyo-danielc** [comment]
The "No route to host" exception gets raised on Linux and macOS when there is no IPv6 connectivity on the system, I've put a fix on the 2024.11 branch, please re-open this issue if it isn't resolved for you in the next beta.

**Comment by Emc1923** [comment]
Closing as per the above comment.
